src/transit.js

```diff
@@ -337,17 +339,44 @@ class Transit {
 			return;
 		}
 
-		// Remove pending request
-		this.removePendingRequest(id);
-
 		this.logger.debug(`Response '${req.action.name}' received from '${packet.sender}'.`);
 
 		// Update nodeID in context (if it uses external balancer)
 		req.ctx.nodeID = packet.sender;
 
 		// Merge response meta with original meta
 		_.assign(req.ctx.meta, packet.meta);
-
+		//Stream case
+		//get the underlined stream for id
+		if(packet.stream !== undefined && packet.stream !== "undefined"){
```

Comment:
Is it necessary? `packet.stream` is defined as boolean. So it can be `true`, `false` or `undefined` if the other side is not supported it. So I think it can be just `if (packet.stream && packet.data)`

Reply:
In my thought if `stream === undefined` that means the response has no stream , if we do just `packet.stream && packet.data` when to close the stream since `false` and `undefined` will be treated as the same ! 

I think we could just do `if(packet.stream !== undefined){` since stream is boolean like you said.

Reply:
Ok, I see.  `if (packet.stream !== undefined) {` will be good.